prime-router/metadata/fhir_mapping/hl7/segments/MSH/MessageHeader.yml

```diff
@@ -28,98 +49,224 @@ sender:
+      condition: $msh6 NOT_NULL && $msh23 NOT_NULL
+      constants:
+        xonOrganizationHL7Use: MSH.23
+      expressionsMap:
+        receiver:
+          expressionType: reference
+          valueOf: datatypes/XON/Organization
+          specs: MSH.23
 
-responsible:
+language:
   vars:
-    msh22: MSH.22
-  condition: $msh22 NOT_NULL
-  expressionType: reference
-  valueOf: datatypes/XON/Organization
-  specs: MSH.22
+    cwe1: MSH.19.1
+  expressionType: nested
+  expressions:
+    - condition: $cwe1 NOT_NULL
+      valueOf: MSH.19.1
+      type: STRING
+      expressionType: HL7Spec
+    - condition: $cwe1 NULL
+      valueOf: MSH.19.4
+      type: STRING
+      expressionType: HL7Spec
 
-extension_2:
-  condition: $msh15 NOT_NULL
+
+# MSH.10 is captured in Bundle.Identifier
+# MSH.12 is hardcoded to 2.5.1
+extension_messageHeader:
   generateList: true
   expressionType: nested
-  vars:
-    msh15: MSH.15
-  expressionsMap:
-    url:
-      type: SYSTEM_URL
-      value: accept-acknowledgement-type
-    valueIdentifier:
-      valueOf: datatype/Identifier_SystemID
+  expressions:
+    - condition: $msh2 NOT_NULL && $msh2 EQUALS_STRING ^~\&#
+      expressionType: nested
+      vars:
+        msh2: String, MSH.2
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: encoding-characters
+        valueString:
+          type: STRING
+          valueOf: MSH.2
+          expressionType: HL7Spec
+    - expressionType: nested
+      vars:
+        msh7: MSH.7
+      condition: $msh7 NOT_NULL
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: msh-7-datetime-of-message
+        valueString:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: MSH.7
+    - expressionType: nested
+      vars:
+        msh13: MSH.13
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: msh-13-sequence-number
+        valueString:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: MSH.13
+    - expressionType: nested
+      vars:
+        msh14: MSH.14
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: msh-14-continuation-pointer
+        valueString:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: MSH.14
+    - condition: $msh15 NOT_NULL
+      expressionType: nested
+      vars:
+        msh15: MSH.15
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: accept-acknowledgement-type
+        valueString:
+          valueOf: MSH.15
+          expressionType: HL7Spec
+          type: STRING
+    - condition: $msh16 NOT_NULL
+      expressionType: nested
+      vars:
+        msh16: MSH.16
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: application-acknowledgement-type
+        valueString:
+          type: STRING
+          valueOf: MSH.16
+          expressionType: HL7Spec
```

Comment:
Why are the field names(msh-15, msh-16) missing from the extension names on these two?

Reply:
These were existing extensions and I adopted not to adjust them

Reply:
More than the hl7-use values, I feel like we should keep a naming pattern on these since they'll be public and potentially used by senders one day. At the least they'll be included in documentation and if we have a different naming pattern we'll eventually get a question from a designer asking why these are different.

Reply:
Looks like SR isn't using this extension so it shouldn't be an issue to change